samples/bluetooth/peripheral_hids/src/main.c

```diff
@@ -72,8 +74,8 @@ static void security_changed(struct bt_conn *conn, bt_security_t level,
 	if (!err) {
 		printk("Security changed: %s level %u\n", addr, level);
 	} else {
-		printk("Security failed: %s level %u err %d\n", addr, level,
-		       err);
+		printk("Security failed: %s level %u err %s(%d)\n", addr, level,
+		       bt_security_err_to_str(err), err);
```

Comment:
To avoid increasing code size too much, we would need conditional compilation of error strings like we do in https://github.com/zephyrproject-rtos/zephyr/pull/73795